python_snippets/using_ast_to_extract_docstrings.py

```diff
@@ -0,0 +1,44 @@
+import ast
+
+
+def load_source_code(source_filename=__file__):
+    """
+    Open a script to parse.
+​
+    :param source_filename: Full path to script to parse, if not provided,
+                            this script will parse itself.
+    :return: Script content as a string
+    """
+    return open(source_filename, 'r').read()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Analysis of this code determined that this line of code contains a resource that might not have closed properly. A resource leak can slow down or crash your system. Programs are strongly recommended to use the built in `with` keyword to open a resource or to use a `try-finally` block to open and close resources explicitly. The [contextlib](https://docs.python.org/3/library/contextlib.html) module provides helpful utilities for using the `with` statement.

[Learn more](https://docs.python.org/3/tutorial/inputoutput.html#reading-and-writing-files)